File: dart/behavioral/mediator/guru_example/bin/component2.dart
import 'base_component.dart';
import 'mediator.dart';

class Component2 extends BaseComponent {
  Component2({Mediator m}) : super();

  void doC() {
    print('Component2 doC.');
    mediator.notify(this, 'C');
  }

  void doD() {
    print('Component2 doD.');
    mediator.notify(this, 'D');
  }
}

File: typescript/behavioral/mediator/guru_example/component2.ts
import {BaseComponent} from "./baseComponent";

export class Component2 extends BaseComponent {
  public doC(): void {
    console.log('Component2 doC.')
    this.mediator.notify(this, 'C')
  }

  public doD(): void {
    console.log('Component2 doD.')
    this.mediator.notify(this, 'D')
  }
}
